docker/README.md

```diff
@@ -1,26 +1,90 @@
+ - myadmin : phpmyadmin accessible on `localhost:8086`
+ - mail : container to intercept email send by yeswiki. Webmail is accessible on `localhost:1080`.You have to set the following in `wakka.config.php`
+ ```
+'contact_mail_func' => 'smtp',
+'contact_smtp_host' => 'mail',
+'contact_smtp_port' => '1025',
+ ```
+### How-To
+
+> [!]info
+> all commands have to be launched from docker directory
+
+To be able to develop locally without messing up with users and permissions, the dev dockerfile uses the same user and group as computer user.
+You need to create a file called `.env` within the `docker` directory with the following content : 
+```
+export UID="YOUR_USER_ID" # can be found with id -u
+export GID="YOUR_USER_GID" # can be found with id -g
+```
+
+Then you can build the container with the following command : 
+```
 docker compose build
 ```
 
-## Launch image
+Once done, you can start containers : 
+```
+docker compose up
+# or docker compose up -d if you want to detach from terminal
+```
+
+It should take some time for the first launch, it will perform `compose install` and `yarn install`.
+Then yeswiki will be accessible at [localhost:8085](http://localhost:8085),
+ phpmyadmin at [localohost:8086](http://localhost:8086) and mailcatcher at [localhost:1080](http://localhost:1080).
+
+Once on the install page, use the following values : 
+
+- **Mysql server host** : yeswiki-db
+- **MYSQL database name** : yeswiki (can be fond in yeswiki.secret)
+- **MYSQL username** : yeswiki (can be fond in yeswiki.secret)
+- **MYSQL password** : password (can be fond in yeswiki.secret)
 
-- `docker compose up -d`
-- yeswiki should be accessible at `localhost:8085`
+> [!]tips
+> if you have a previous developpement installation you may need to change value accordingly in the wakka.config.php
 
-## Dev version
+## reinitialize yeswiki repo from dev
 
-- allow www-data to right local directory
-  This version should map the local repository to your docker container.
-- `docker compose up -f docker-compose-dev.yml`
+docker create and populates the following folders files :
 
-## Remove all docker images and volumes
+- vendor (for php dependencies)
+- node_modules (for yarn dependencies)
+- wakka.config.php
+- cache
+- tools/bazar/vendor/
 
-```bash
-docker compose stop
-docker compose rm
-docker volume rm yeswiki yeswiki-db
+It should be enough to remove the `wakka.config.php` file
+
+## Remove database
+
+- remove containers (stopping container doesn't remove them)
+```
+docker compose down 
+```
+- remove docker volume (containing database files)
+```
+docker volume rm yeswiki-db
+```
+
+## updating php or yarn dependency
+
+You can simply restart container with the following command : 
+```
+docker compose restart
+```
+
+If you want to update php or yarn dependency without restarting everything, you can do the following commands.
+```
+docker compose exec yeswiki-app composer install
+docker compose exec yeswiki-app yarn install
```

Comment:
Chez moi ces commandes ne fonctionnent pas

`OCI runtime exec failed: exec failed: unable to start container process: exec: "yarn install": executable file not found in $PATH: unknown`

Aussi, si je me connecte au container avec un `docker compose exec yeswiki-web bash` et que ensuite je fait un yarn ou un composer, la commande se lance mais échoue à cause de problèmes de permissions

`yarn install` => `error Error: EACCES: permission denied, unlink '/var/www/html/node_modules/@babel/parser/CHANGELOG.md'`

`composer install` => `In Filesystem.php line 288:=Could not delete /var/www/html/vendor/bin/carbon:  `
                                                    
alors que dpeuis le entrypoint ça fonctionne. (Ps: j'ai bien créé un docker/.env avec mon user id & gorup)